### PR TITLE
Initialize nonce manager.

### PR DESCRIPTION
**Pull request by winder**
Add bindings for the NonceManager and finish implementing the CCIPReader.

### CODE REVIEW

File: internal/reader/ccip.go [hunk 1]
```diff
@@ -476,10 +479,52 @@ func (r *CCIPChainReader) Sync(ctx context.Context) (bool, error) {
...
+		return fmt.Errorf("get onramps: %w", err)
+	}
+
+	if _, ok := r.contractReaders[r.destChain]; !ok {
+		r.lggr.Debugw("skipping nonce manager, dest chain not configured for this deployment",
+			"destChain", r.destChain)
+		return nil
+	}
+
+	// Bind the onRamp contract address to the reader.
```
Comment (asoliman92) [inline internal/reader/ccip.go:501]:
```suggestion
	// Bind the nonceManager contract address to the reader.
```

File: internal/reader/ccip.go [hunk 2]
```diff
@@ -557,5 +602,33 @@ func (r *CCIPChainReader) validateWriterExistence(chains ...cciptypes.ChainSelec
...
+		map[string]any{},
+		&resp,
+	)
+	if err != nil {
+		return offrampStaticChainConfig{}, fmt.Errorf("failed to get source chain config: %w", err)
+	}
+	return resp, nil
+}
+
+type offrampStaticChainConfig struct {
```
Comment (asoliman92) [inline internal/reader/ccip.go:626]: Can you put link to the offramp type? Will make it easier to find it.

Comment (winder) [reply internal/reader/ccip.go:626]: Yeah, I'll add a comment, what did you have in mind? A comment? I copied the pattern used for `type sourceChainConfig struct {`, so I'll update that as well.

**Review by makramkd — COMMENTED** [review]
You can write an integration test to test that this works in the ccipreader_test.go file in chainlink/ccip. See `TestCCIPReader_CommitReportsGTETimestamp` as an example.

File: internal/reader/ccip.go [hunk 3]
```diff
@@ -523,10 +567,52 @@ func (r *CCIPChainReader) Sync(ctx context.Context) (bool, error) {
...
+			Name:    consts.ContractNameNonceManager,
+		},
+	}); err != nil {
+		return fmt.Errorf("bind nonce manager: %w", err)
+	}
+
+	return nil
+}
+
+func (r *CCIPChainReader) Sync(ctx context.Context) (bool, error) {
```
Comment (makramkd) [inline internal/reader/ccip.go:605]: @dimkouv why the `(bool, error)` return signature? I feel like `error` is enough?

File: internal/reader/ccip.go (hunk 2 above)
Comment (makramkd) [reply internal/reader/ccip.go:715]: A permalink to the solidity struct would be good enough IMO

**Review by makramkd — APPROVED** [review]

**Pull request merged**